v8.go

```diff
@@ -163,7 +163,22 @@ func NewIsolate() *Isolate {
 // to initialize all Contexts created from this Isolate.
 func NewIsolateWithSnapshot(s *Snapshot) *Isolate {
 	v8_init_once.Do(func() { C.v8_init() })
-	iso := &Isolate{ptr: C.v8_Isolate_New(s.data), s: s}
+	iso := &Isolate{ptr: C.v8_Isolate_New(s.data, nil), s: s}
+	runtime.SetFinalizer(iso, (*Isolate).release)
+	return iso
+}
+
+type ResourceConstraints struct {
+	// MaxOldSpaceSize sets the maximum size of the old object heap in MiB.
+	MaxOldSpaceSize int
+}
+
+// NewIsolateWithConstraints creates a new V8 Isolate applying additional
+// resource constraints to limit the V8 runtime.
+func NewIsolateWithConstraints(constraints ResourceConstraints) *Isolate {
```

Comment:
In retrospect, the `NewIsolateWithSnapshot` was a mistake -- we should have `NewIsolateWithOptions` so that every new constructor option doesn't add another constructor.

Reply:
Yes. Is it too late for that given that the existing API shouldn't be broken?

Reply:
It's too late to remove `NewIsolateWithSnapshot`, but we can add `NewIsolateWithOptions` and stem the bleeding.  Add a note that `NewIsolateWithSnapshot` is deprecated and just forward the implementation to a call to `...WithOptions`